Improve test expressivity

BasThomas opened the pull request:
Using specific assert functions will generate easier to understand failure messages if the tests were to fail. For example, in the `XCTAssertNil` case, it will say "something was unexpectedly not nil" instead of only "test failed".

BasThomas committed:
Improve test expressivity

Using specific assert functions will generate easier to understand failure messages if the tests were to fail. For example, in the `XCTAssertNil` case, it will say "something was unexpectedly not nil" instead of only "test failed".

BasThomas committed:
Remove default arguments

BasThomas commented on MockerTests/MockerTests.swift:
```diff
@@ -290,7 +290,7 @@ final class MockerTests: XCTestCase {
         let urlRequest = URLRequest(url: simpleURL!)
-        XCTAssertEqual(composedURL.absoluteString, simpleURL!.absoluteString)
+        XCTAssertEqual(composedURL.absoluteString, simpleURL?.absoluteString)
         XCTAssert(mock == urlRequest)
```
This one I didn't change; somehow it is comparing a `Mock` to a `URLRequest`... but `XCTAssertEqual` didn't seem to be able to figure out the generics, so I left it as-is.

wetransferplatform commented on MockerTests/MockerTests.swift:
```diff
@@ -42,11 +42,11 @@ final class MockerTests: XCTestCase {
         
```
<!--
  0 failure: 
  1 warning:  Lines should not ...
  
  
  DangerID: danger-id-Danger;
  File: MockerTests/MockerTests.swift;
  Line: 42;
-->

- :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

wetransferplatform commented on MockerTests/MockerTests.swift:
```diff
@@ -63,11 +63,11 @@ final class MockerTests: XCTestCase {
+            XCTAssertNil(error)
             let image: UIImage = UIImage(data: data!)!
             let sampleImage: UIImage = UIImage(contentsOfFile: MockedData.botAvatarImageFileUrl.path)!
             
```
<!--
  0 failure: 
  1 warning:  Lines should not ...
  
  
  DangerID: danger-id-Danger;
  File: MockerTests/MockerTests.swift;
  Line: 69;
-->

- :warning: Lines should not have trailing whitespace. (`trailing_whitespace`)

BasThomas replied:
Not seeing this one locally :(

wetransferplatform commented on MockerTests/MockerTests.swift:
```diff
@@ -87,14 +87,14 @@ final class MockerTests: XCTestCase {
+            XCTAssertNil(error)
             guard let data = data, let image: UIImage = UIImage(data: data) else {
-                XCTFail("Invalid data")
+                XCTFail("Invalid data \(String(describing: data))")
```
<!--
  1 failure:  variable declared...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: MockerTests/MockerTests.swift;
  Line: 92;
-->
- :no_entry_sign: variable declared in 'guard' condition is not usable in its body

AvdLee replied:
You might want to fix this @basthomas

AvdLee reviewed (approved):
This is a very welcome change! I couldn't agree more, using the right assertions for the job will help a lot 🙂

BasThomas commented on MockerTests/MockerTests.swift:
```diff
@@ -87,14 +87,14 @@ final class MockerTests: XCTestCase {
+            XCTAssertNil(error)
             guard let data = data, let image: UIImage = UIImage(data: data) else {
-                XCTFail("Invalid data")
+                XCTFail("Invalid data \(String(describing: data))")
```
Hm, seems like a false positive, kind of? `data` is used to construct the `image`. Will take a look.

BasThomas replied:
Oooh, now I see. It's complaining about me using the "unavailable" variable within the body. "of course", this is just the nullable `data` we want to use. What do you think? Should I provide SwiftLint exceptions for these?

AvdLee replied:
I would either split the guard so data is available, or not log data at all in the exception I think!

BasThomas replied:
Hm, will split then 👍

kairadiagne reviewed (approved):
Great work,  a small change that makes working with the tests much better!

BasThomas committed:
Split up guards so we can guarantee non-nil data

The SwiftLint configuration would otherwise complain with the following error:

> 🚫 variable declared in 'guard' condition is not usable in its body

wetransferplatform commented:
<!--
  0 failure: 
  1 warning:  Consider to place...
  2 messages
  2 markdown notices
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Consider to place some `MARK:` lines for Sources/Mock.swift, which is over 200 lines big.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/b649bab6-cf70-42b5-9ca6-39a7ebe14a28" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Mocker: Executed 25 tests, with 0 failures (0 unexpected) in 1.864 (1.922) seconds</td>
    </tr>
  </tbody>
</table>

## MockerTests.xctest: Coverage: 95.75
| File | Coverage ||
| --- | --- | --- |
MockerTests.swift | 95.73% | ✅


## Mocker.framework: Coverage: 88.04
| File | Coverage ||
| --- | --- | --- |
MockingURLProtocol.swift | 86.41% | ✅
Mock.swift | 82.18% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 985b218d511dfcfc65bb3882c934c35c2e155028
</p>

AvdLee reviewed (dismissed):
@BasThomas can you have a look at the trailing whitespace warnings?

BasThomas commented:
😢 yes. I sweat I have the setting enabled in Xcode. Wonder if something is broken.

BasThomas commented:
Also weird as I do not seem to have changed them? 🤔

BasThomas committed:
Run `swiftlint autocorrect`

BasThomas commented:
`swiftlint autocorrect` it is.

BasThomas commented:
I don't understand... Bitrise is unhappy, but I can't find the attached run that would have failed :(

BasThomas committed:
Trigger CI

wetransferplatform commented:
Congratulations! :tada: This was released as part of [Release 2.5.4](https://github.com/WeTransfer/Mocker/releases/tag/2.5.4) :rocket:<p align="right"><i>Generated by <a href="https://github.com/WeTransfer/GitBuddy">GitBuddy</a></i></p>

wetransferplatform commented:
Congratulations! :tada: This was released as part of [Release 2.5.4](https://github.com/WeTransfer/Mocker/releases/tag/2.5.4) :rocket:<p align="right"><i>Generated by <a href="https://github.com/WeTransfer/GitBuddy">GitBuddy</a></i></p>
